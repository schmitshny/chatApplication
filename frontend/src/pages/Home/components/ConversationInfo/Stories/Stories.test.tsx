import { describe, expect, test } from 'vitest';
import { Stories } from '.';
import { render } from '../../../../../tests/test-utils';
import { waitFor, screen, fireEvent } from '@testing-library/react';
import { server } from '../../../../../tests/server';
import { HttpResponse, http } from 'msw';
import { mockStories } from '../../../../../features/stories/tests/mocks';

describe('Stories', () => {
  test('should render stories', async () => {
    const { container } = render(<Stories />);

    await waitFor(() => {
      expect(screen.getByText('Stories')).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });

  test('should open story viewer when clicking on a story', async () => {
    const { container } = render(<Stories />);

    await waitFor(() => {
      expect(screen.getByText('Stories')).toBeInTheDocument();
    });

    const firstStory = screen.getAllByTestId('single-story');

    expect(firstStory).toHaveLength(1);

    const event = fireEvent.click(firstStory[0]);

    expect(event).toBeTruthy();

    expect(screen.getByTestId('story-viewer')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  test('should render multiple stories', async () => {
    server.use(
      http.get('http://localhost:8000/stories', () => {
        return HttpResponse.json(mockStories);
      }),
    );

    const { container } = render(<Stories />);

    const stories = await screen.findAllByTestId('single-story');

    expect(stories).toHaveLength(3);

    await waitFor(() => {
      expect(screen.getByText('Stories')).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
  });
});
