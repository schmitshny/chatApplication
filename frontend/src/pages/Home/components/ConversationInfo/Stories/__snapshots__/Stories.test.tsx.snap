// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stories > should open story viewer when clicking on a story 1`] = `
<div>
  <div
    class="sc-himrge hMtlhF"
  >
    <h4
      class="sc-gXmRPE jtLRgf"
    >
      Stories
    </h4>
    <ul
      class="sc-cCsQgl khRwEb"
    >
      <div
        class="sc-bZkfhe jitLc"
      >
        <div
          class="sc-jSMdHm eJItaO"
        >
          <img
            alt="user avatar"
            class="sc-gKXNOb gUEVCF"
            loading="lazy"
            shape="circle"
            src="/src/components/Avatar/assets/defaultAvatar.jpg"
          />
        </div>
        <div
          class="sc-kLLYlN bhvsNN"
        >
          <div
            class="sc-jqUWZQ hkKBeh"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
                stroke="#1C274C"
                stroke-width="1.5"
              />
              <path
                d="M15 12L12 12M12 12L9 12M12 12L12 9M12 12L12 15"
                stroke="#36de36
"
                stroke-linecap="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-bZkfhe jitLc"
        data-testid="single-story"
      >
        <div
          class="sc-jSMdHm eJItaO"
        >
          <img
            alt="user avatar"
            class="sc-gKXNOb gUEVCF"
            loading="lazy"
            shape="circle"
            src="http://example.com/story1.jpg"
          />
        </div>
      </div>
    </ul>
    <div
      class="sc-ksZauW iVoqlh"
    >
      <div
        class="sc-hAZoWV igRyVY"
      >
        <div
          class="sc-ciZfDS enGGdO"
          data-testid="story-viewer"
        >
          <button
            class="sc-iTOPbJ sc-iAvezq kLjlyc jCAHyq"
            data-testid="next-button"
            disabled=""
          >
            <div
              class="sc-jqUWZQ bnkMjB"
            >
              <svg
                fill="none"
                height="34"
                id="Layer_1"
                version="1.1"
                viewBox="0 0 512 512"
                width="34"
                xml:space="preserve"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g>
                  <g>
                    <path
                      d="M256,0C114.837,0,0,114.837,0,256s114.837,256,256,256s256-114.837,256-256S397.163,0,256,0z M313.749,347.584
			c8.341,8.341,8.341,21.824,0,30.165c-4.16,4.16-9.621,6.251-15.083,6.251c-5.461,0-10.923-2.091-15.083-6.251L176.917,271.083
			c-8.341-8.341-8.341-21.824,0-30.165l106.667-106.667c8.341-8.341,21.824-8.341,30.165,0s8.341,21.824,0,30.165L222.165,256
			L313.749,347.584z"
                    />
                  </g>
                </g>
              </svg>
            </div>
          </button>
          <div
            class="sc-jdANtv jCCKIa"
          >
            <div
              class="sc-dPyBWt bgdDBd"
            />
            <div
              class="sc-cTQgWk hcgoYM"
            >
              <div
                class="sc-jSMdHm eJItaO"
              >
                <img
                  alt="user avatar"
                  class="sc-gKXNOb gUEVCF"
                  loading="lazy"
                  shape="circle"
                  src="http://example.com/avatar.jpg"
                />
              </div>
              <span
                class="sc-jOrOLN gzCRYg"
              >
                John Doe
              </span>
            </div>
            <img
              alt="Story"
              class="sc-bZnhmg hYXLeG"
              src="http://example.com/story1.jpg"
            />
          </div>
          <button
            class="sc-iTOPbJ sc-efBcPX kLjlyc meewA"
            data-testid="previous-button"
            disabled=""
          >
            <div
              class="sc-jqUWZQ bnkMjB"
            >
              <svg
                fill="none"
                height="34"
                id="Layer_1"
                version="1.1"
                viewBox="0 0 512 512"
                width="34"
                xml:space="preserve"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g>
                  <g>
                    <path
                      d="M256,0C114.837,0,0,114.837,0,256s114.837,256,256,256s256-114.837,256-256S397.163,0,256,0z M313.749,347.584
			c8.341,8.341,8.341,21.824,0,30.165c-4.16,4.16-9.621,6.251-15.083,6.251c-5.461,0-10.923-2.091-15.083-6.251L176.917,271.083
			c-8.341-8.341-8.341-21.824,0-30.165l106.667-106.667c8.341-8.341,21.824-8.341,30.165,0s8.341,21.824,0,30.165L222.165,256
			L313.749,347.584z"
                    />
                  </g>
                </g>
              </svg>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stories > should render multiple stories 1`] = `
<div>
  <div
    class="sc-himrge hMtlhF"
  >
    <h4
      class="sc-gXmRPE jtLRgf"
    >
      Stories
    </h4>
    <ul
      class="sc-cCsQgl khRwEb"
    >
      <div
        class="sc-bZkfhe jitLc"
      >
        <div
          class="sc-jSMdHm eJItaO"
        >
          <img
            alt="user avatar"
            class="sc-gKXNOb gUEVCF"
            loading="lazy"
            shape="circle"
            src="/src/components/Avatar/assets/defaultAvatar.jpg"
          />
        </div>
        <div
          class="sc-kLLYlN bhvsNN"
        >
          <div
            class="sc-jqUWZQ hkKBeh"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
                stroke="#1C274C"
                stroke-width="1.5"
              />
              <path
                d="M15 12L12 12M12 12L9 12M12 12L12 9M12 12L12 15"
                stroke="#36de36
"
                stroke-linecap="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-bZkfhe jitLc"
        data-testid="single-story"
      >
        <div
          class="sc-jSMdHm eJItaO"
        >
          <img
            alt="user avatar"
            class="sc-gKXNOb gUEVCF"
            loading="lazy"
            shape="circle"
            src="http://example.com/story1.jpg"
          />
        </div>
      </div>
      <div
        class="sc-bZkfhe jitLc"
        data-testid="single-story"
      >
        <div
          class="sc-jSMdHm eJItaO"
        >
          <img
            alt="user avatar"
            class="sc-gKXNOb gUEVCF"
            loading="lazy"
            shape="circle"
            src="http://example.com/story2.jpg"
          />
        </div>
      </div>
      <div
        class="sc-bZkfhe jitLc"
        data-testid="single-story"
      >
        <div
          class="sc-jSMdHm eJItaO"
        >
          <img
            alt="user avatar"
            class="sc-gKXNOb gUEVCF"
            loading="lazy"
            shape="circle"
            src="http://example.com/story3.jpg"
          />
        </div>
      </div>
    </ul>
  </div>
</div>
`;

exports[`Stories > should render stories 1`] = `
<div>
  <div
    class="sc-himrge hMtlhF"
  >
    <h4
      class="sc-gXmRPE jtLRgf"
    >
      Stories
    </h4>
    <ul
      class="sc-cCsQgl khRwEb"
    >
      <div
        class="sc-bZkfhe jitLc"
      >
        <div
          class="sc-jSMdHm eJItaO"
        >
          <img
            alt="user avatar"
            class="sc-gKXNOb gUEVCF"
            loading="lazy"
            shape="circle"
            src="/src/components/Avatar/assets/defaultAvatar.jpg"
          />
        </div>
        <div
          class="sc-kLLYlN bhvsNN"
        >
          <div
            class="sc-jqUWZQ hkKBeh"
          >
            <svg
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
                stroke="#1C274C"
                stroke-width="1.5"
              />
              <path
                d="M15 12L12 12M12 12L9 12M12 12L12 9M12 12L12 15"
                stroke="#36de36
"
                stroke-linecap="round"
                stroke-width="1.5"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="sc-bZkfhe jitLc"
        data-testid="single-story"
      >
        <div
          class="sc-jSMdHm eJItaO"
        >
          <img
            alt="user avatar"
            class="sc-gKXNOb gUEVCF"
            loading="lazy"
            shape="circle"
            src="http://example.com/story1.jpg"
          />
        </div>
      </div>
    </ul>
  </div>
</div>
`;
